### PR TITLE
✨add `stock-status` values in the `valueFacets` aggregation

### PR DESCRIPTION
**Pull request by gengjozsef**
- Implement `StockStatusResultFormatterPlugin`
- Add expand query with the `stock-status` aggregation
- Extend Unit Tests
- Extend `transfer` objects

### CODE REVIEW

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    /**
+     * @param \Spryker\Client\Kernel\Container $container
+     *
+     * @return \Spryker\Client\Kernel\Container
+     */
+    public function provideServiceLayerDependencies(Container $container)
+    {
+        $container = $this->addCustomerClient($container);
+
+        return $container;
```
Comment (daniel-rose) [inline bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php:28]:
```suggestion
        $container = parent::provideServiceLayerDependencies($container);

        return $this->addCustomerClient($container);
```

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    /**
+     * @param \Spryker\Client\Kernel\Container $container
+     *
+     * @return \Spryker\Client\Kernel\Container
+     */
+    protected function addCustomerClient(Container $container)
```
Comment (daniel-rose) [inline bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php:36]:
```suggestion
    protected function addCustomerClient(Container $container): Container
```

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/Transfer/conditional_availability_product_page_search.transfer.xml [hunk 3]
```diff
@@ -23,4 +23,12 @@
...
     </transfer>
 
+    <transfer name="RestCatalogSearchAttributes">
+        <property name="stockStatus" type="RestStockStatusSearchResult[]" singular="stockStatus"/>
+    </transfer>
+
+    <transfer name="RestStockStatusSearchResult">
+        <property name="values" type="array" singular="value"/>
+    </transfer>
+
```
Comment (daniel-rose) [inline bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/Transfer/conditional_availability_product_page_search.transfer.xml:33]: Obsolete?

**Review by daniel-rose — CHANGES_REQUESTED** [review]

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php (hunk 1 above)
Comment (gengjozsef) [reply bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php]: @daniel-rose  i've made the suggested changes

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php (hunk 2 above)
Comment (gengjozsef) [reply bundles/conditional-availability-product-page-search/src/FondOfImpala/Client/ConditionalAvailabilityProductPageSearch/ConditionalAvailabilityProductPageSearchDependencyProvider.php]: @daniel-rose i've made the suggested changes

File: bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/Transfer/conditional_availability_product_page_search.transfer.xml (hunk 3 above)
Comment (gengjozsef) [reply bundles/conditional-availability-product-page-search/src/FondOfImpala/Shared/ConditionalAvailabilityProductPageSearch/Transfer/conditional_availability_product_page_search.transfer.xml]: @daniel-rose i've removed the obsolate code

**Review by daniel-rose — APPROVED** [review]

**Pull request merged**